book/src/prefabs/prefabs_technical_explanation.md

```diff
@@ -0,0 +1,323 @@
+parameter to the function. This parameter contains a list of all entities affected by the `Prefab`, the first
+entry in the list will be the main `Entity`, and the rest will be the entities that were created for all the
+entries in the data list inside the `Prefab`.
+
+Last of all, we can see that this does not implement `load_sub_assets`, which is because there
+are no secondary assets to load from `Source` here.
+
+Let's look at a slightly more complex implementation, the `AssetPrefab`. This `PrefabData` is used to
+load extra `Asset`s as part of a `Prefab`:
+
+```rust,no_run,noplaypen
+# extern crate amethyst;
+# #[macro_use] extern crate serde_derive;
+# use amethyst::assets::{Asset, AssetStorage, Loader, Format, Handle, ProgressCounter};
+# use amethyst::assets::PrefabData;
+# use amethyst::ecs::{WriteStorage, ReadExpect, Read, Entity, error::Error as SpecsError};
+#
+#[derive(Deserialize, Serialize)]
+pub enum AssetPrefab<A, F>
+where
+    A: Asset,
+    F: Format<A>,
+{
+    /// From existing handle
+    #[serde(skip)]
+    Handle(Handle<A>),
+
+    /// From file, (name, format, format options)
+    File(String, F, F::Options),
+}
+
+impl<'a, A, F> PrefabData<'a> for AssetPrefab<A, F>
+where
+    A: Asset,
+    F: Format<A> + Clone,
+    F::Options: Clone,
+{
+    type SystemData = (
+        ReadExpect<'a, Loader>,
+        WriteStorage<'a, Handle<A>>,
+        Read<'a, AssetStorage<A>>,
+    );
+
+    type Result = Handle<A>;
+
+    fn add_to_entity(
+        &self,
+        entity: Entity,
+        system_data: &mut Self::SystemData,
+        _: &[Entity],
+    ) -> Result<Handle<A>, SpecsError> {
+        let handle = match *self {
+            AssetPrefab::Handle(ref handle) => handle.clone(),
+            AssetPrefab::File(ref name, ref format, ref options) => system_data.0.load(
+                name.as_ref(),
+                format.clone(),
+                options.clone(),
+                (),
+                &system_data.2,
+            ),
+        };
+        system_data.1.insert(entity, handle.clone()).map(|_| handle)
+    }
+
+    fn load_sub_assets(
+        &mut self,
+        progress: &mut ProgressCounter,
+        system_data: &mut Self::SystemData,
+    ) -> Result<bool, SpecsError> {
+        let handle = match *self {
+            AssetPrefab::File(ref name, ref format, ref options) => Some(system_data.0.load(
+                name.as_ref(),
+                format.clone(),
+                options.clone(),
+                progress,
+                &system_data.2,
+            )),
+            _ => None,
+        };
+        if let Some(handle) = handle {
+            *self = AssetPrefab::Handle(handle);
+        }
+        Ok(true)
+    }
+}
+```
+
+So, there are two main differences to this `PrefabData` compared the `Transform` example.
+The first difference is that the `add_to_entity` function now return a `Handle<A>`.
```

Comment:
What happen to the returned Handle<A>? Since it's the PrefabLoaderSystem  that call the add_to_entity function, I may need to know what it does with it.